webforj-addons-components/webforj-map/src/main/java/com/webforj/addons/components/map/Map.java

```diff
@@ -0,0 +1,197 @@
+   * @return this Map instance for method chaining
+   */
+  public Map setCenter(Coordinate coordinate) {
+    Objects.requireNonNull(coordinate, "Coordinate cannot be null");
+    self.callJsFunctionVoidAsync("setCenter", coordinate.toArray());
+    return this;
+  }
+
+  /**
+   * Sets the map center using longitude and latitude.
+   *
+   * @param longitude the longitude in degrees
+   * @param latitude the latitude in degrees
+   * @return this Map instance for method chaining
+   */
+  public Map setCenter(double longitude, double latitude) {
+    return setCenter(new Coordinate(longitude, latitude));
+  }
+
+  /**
+   * Sets the map zoom level.
+   *
+   * @param zoom the zoom level
+   * @return this Map instance for method chaining
+   */
+  public Map setZoom(int zoom) {
+    if (zoom < 0) {
+      throw new IllegalArgumentException("Zoom level cannot be negative: " + zoom);
+    }
+    self.callJsFunctionVoidAsync("setZoom", zoom);
+    return this;
+  }
+
+  /**
+   * Sets the map rotation in radians.
+   *
+   * @param rotation the rotation angle in radians
+   * @return this Map instance for method chaining
+   */
+  public Map setRotation(double rotation) {
+    self.callJsFunctionVoidAsync("setRotation", rotation);
+    return this;
+  }
+
+  /**
+   * Fits the map view to the specified extent.
+   *
+   * @param extent the extent to fit
+   * @return this Map instance for method chaining
+   */
+  public Map setExtent(Extent extent) {
+    Objects.requireNonNull(extent, "Extent cannot be null");
+    self.callJsFunctionVoidAsync("setExtent", extent.toArray(), 0);
+    return this;
+  }
+
+  /**
+   * Fits the map view to the specified extent with padding.
+   *
+   * @param extent the extent to fit
+   * @param padding padding around the extent in pixels
+   * @return this Map instance for method chaining
+   * @throws IllegalArgumentException if padding is negative
+   */
+  public Map setExtent(Extent extent, int padding) {
+    Objects.requireNonNull(extent, "Extent cannot be null");
+    if (padding < 0) {
+      throw new IllegalArgumentException("Padding cannot be negative: " + padding);
+    }
+    self.callJsFunctionVoidAsync("setExtent", extent.toArray(), padding);
+    return this;
+  }
+
+  /**
+   * Animates the view with the specified options. Use this method to animate center, zoom, or
+   * rotation changes with a custom duration.
+   *
+   * @param options the animation options containing center, zoom, rotation, and duration
+   * @return this Map instance for method chaining
+   */
+  public Map viewAnimate(ViewAnimationOptions options) {
+    Objects.requireNonNull(options, "ViewAnimationOptions cannot be null");
+    self.callJsFunctionVoidAsync("viewAnimate", options);
+    return this;
+  }
+
+  /**
+   * Gets the current center coordinate of the view.
+   *
+   * @return a pending result containing the center coordinate
+   */
+  public PendingResult<Coordinate> getCenter() {
```

Comment:
why these methods are using PendingResult if the client methods are not async? This is complicating the API